src/commands/neuron_permission.rs

```diff
@@ -0,0 +1,93 @@
+use anyhow::anyhow;
+use candid::Encode;
+use clap::{ArgEnum, Parser};
+use ic_base_types::PrincipalId;
+use ic_sns_governance::pb::v1::{
+    manage_neuron::{AddNeuronPermissions, Command, RemoveNeuronPermissions},
+    ManageNeuron, NeuronPermissionList,
+};
+
+use crate::{
+    lib::{
+        parse_neuron_id,
+        signing::{sign_ingress_with_request_status_query, IngressWithRequestId},
+        AnyhowResult, TargetCanister,
+    },
+    SnsCanisterIds,
+};
+
+/// Signs a ManageNeuron message to add or remove permissions for a principal to/from a neuron.
+/// This will selectively enable/disable that principal to do a variety of management tasks for the neuron, including voting and disbursing.
+#[derive(Parser)]
+pub struct NeuronPermissionOpts {
+    /// Whether to add or remove permissions.
+    #[clap(arg_enum)]
+    subcommand: Subcmd,
+    /// The id of the neuron to configure as a hex encoded string. For example:
+    /// 83a7d2b12f654ff58335e5a2512ccae0d7839c744b1807a47c96f5b9f3969069
+    neuron_id: String,
+    /// The principal to change the permissions of.
+    #[clap(long)]
+    principal: PrincipalId,
+    /// The permissions to add to/remove from the principal. You can specify multiple in one command.
+    #[clap(
+        long,
+        multiple_values(true),
+        use_value_delimiter(true),
+        arg_enum,
+        min_values(1),
+        required(true)
+    )]
+    permissions: Vec<PermissionType>,
+}
+
+#[derive(ArgEnum, Clone)]
+pub enum PermissionType {
+    ConfigureDissolveState = 1,
+    ManagePrincipals = 2,
+    SubmitProposal = 3,
+    Vote = 4,
+    Disburse = 5,
+    Split = 6,
+    MergeMaturity = 7,
+    DisburseMaturity = 8,
+}
```

Comment:
Is there any way to keep this in sync with the NeuronPermissionType defined in sns/governance? It would be bad if this was out of sync. I just tested adding this annotation to our potobuf definition of NeuronPermissionType. If this sounds like a good idea to you I can cut an MR adding it. Let me know what you think.

Reply:
If the generated version can have a derive(ArgEnum) then yes, that'd work.

Reply:
Word. MR incoming

Reply:
https://gitlab.com/dfinity-lab/public/ic/-/merge_requests/6126